tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py

```diff
@@ -2215,3 +2206,37 @@ def cat(X):
                 raise AttributeError(e)
             else:
                 raise Exception(e)
+
+    @unittest.skipIf(
+        LooseVersion(sklearn.__version__) < "0.20",
+        reason="columntransformer introduction in 0.20.0",
+    )
+    def test_setupid_with_column_transformer(self):
+        """Test to check if inclusion of ColumnTransformer in a pipleline is treated as a new
+        flow each time.
+        """
+        import sklearn.compose
+        from sklearn.svm import SVC
+
+        def column_transformer_pipe(task_id):
+            task = openml.tasks.get_task(task_id)
+            # make columntransformer
+            preprocessor = sklearn.compose.ColumnTransformer(
+                transformers=[
+                    ("num", StandardScaler(), cont),
+                    ("cat", OneHotEncoder(handle_unknown="ignore"), cat),
+                ]
+            )
+            # make pipeline
+            clf = SVC(gamma="scale", random_state=1)
+            pipe = make_pipeline(preprocessor, clf)
+            # run task
+            run = openml.runs.run_model_on_task(pipe, task, avoid_duplicate_runs=True)
```

Comment:
This seems to fail on several jobs (but strangely not all-but-one, perhaps due to race conditions?). And shouldn't that be correct? 
There is no part of this pipeline which has dynamic setup, which means after the first run upload [isn't it correct to raise an error](https://github.com/openml/openml-python/pull/982/checks?check_run_id=1347584596#step:8:883)?

Reply:
That's a good point that I missed. 
I think for now the correct approach might be to mark the run/setup to me removed using the `_mark_entity_for_removal` feature
I'll push with that change and see the tests and then ask for a review if it passes!

Reply:
Also, @mfeurer is cleaning the test servers, and he recommended waiting this week out before judging if this error is a result of those changes. Nevertheless, I made the push with the change that should be there in any case.

Reply:
Okay we'll wait.

>  if this error is a result of those changes. 

I imagine the times it _did_ pass are those where a previous upload had been deleted by the clean up script in between.

Reply:
I just had a look at this PR and it appears that this error and an error building the docs are the only things that hold back merging this PR? @Neeratyoy could you please have a look into this?